Build fails on target wasm32-unknown-unknown with: cannot find function `get_granularity_helper` in this scope

Here is the full out put:

```bash
(base) [bobajeff@bobajeff-pc page_size_rs]$ cargo build --target wasm32-unknown-unknown --verbose
    Updating crates.io index
   Compiling page_size v0.4.2 (/home/bobajeff/Rust/webassembly/pagesize/page_size_rs)
     Running `rustc --crate-name page_size src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=532c90a805fb4ddc -C extra-filename=-532c90a805fb4ddc --out-dir /home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/deps --target wasm32-unknown-unknown -C incremental=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/incremental -L dependency=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/deps -L dependency=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/debug/deps`
error[E0425]: cannot find function `get_granularity_helper` in this scope
  --> src/lib.rs:65:5
   |
65 |     get_granularity_helper()
   |     ^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

warning: unused import: `std::sync::Once`
  --> src/lib.rs:36:5
   |
36 | use std::sync::Once;
   |     ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0425`.
error: could not compile `page_size`

Caused by:
  process didn't exit successfully: `rustc --crate-name page_size src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=532c90a805fb4ddc -C extra-filename=-532c90a805fb4ddc --out-dir /home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/deps --target wasm32-unknown-unknown -C incremental=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/incremental -L dependency=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/wasm32-unknown-unknown/debug/deps -L dependency=/home/bobajeff/Rust/webassembly/pagesize/page_size_rs/target/debug/deps` (exit code: 1)
(base) [bobajeff@bobajeff-pc page_size_rs]$ 
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -261,3 +261,23 @@ matrix:
     #   nightly 
     - env: TARGET=asmjs-unknown-emscripten
       rust: nightly 
+    # WASI
+    #   stable
+    - env: TARGET=wasm32-wasi
+      rust: stable
+    #   beta
+    - env: TARGET=wasm32-wasi
+      rust: beta
+    #   nightly
+    - env: TARGET=wasm32-wasi
+      rust: nightly
+    # WebAssembly
+    #   stable
+    - env: TARGET=wasm32-unknown-unknown
+      rust: stable
+    #   beta
+    - env: TARGET=wasm32-unknown-unknown
+      rust: beta
+    #   nightly
+    - env: TARGET=wasm32-unknown-unknown
+      rust: nightly

--- a/README.md
+++ b/README.md
@@ -1,4 +1,4 @@
-`page_size_rs` is a Rust crate that provides an easy, fast, cross-platform way to retrieve the memory page size of the current system. It supports any POSIX-compliant system and Windows.
+`page_size_rs` is a Rust crate that provides an easy, fast, cross-platform way to retrieve the memory page size of the current system. It supports any POSIX-compliant system, Windows, and WebAssembly.
 
 [Documentation](https://docs.rs/page_size)
 
@@ -25,7 +25,7 @@ fn main() {
 
 # Platforms
 
-`page_size_rs` should Work on Windows and any POSIX compatible system (Linux, Mac OSX, etc.).
+`page_size_rs` should Work on Windows, any POSIX compatible system (Linux, Mac OSX, etc.), and WebAssembly.
 
 `page_size_rs` is continuously tested on:
   * `x86_64-unknown-linux-gnu` (Linux)
@@ -54,3 +54,5 @@ fn main() {
   * `x86_64-unknown-freebsd`
   * `x86_64-unknown-netbsd`
   * `asmjs-unknown-emscripten`
+  * `wasm32-wasi`
+  * `wasm32-unknown-unknown`

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -111,7 +111,7 @@ mod unix {
 
 // WebAssembly does not have a specific allocation granularity.
 // The page size works well.
-#[cfg(any(target_arch = "wasm32", target_arch = "wasm64"))]
+#[cfg(all(not(target_os = "emscripten"), any(target_arch = "wasm32", target_arch = "wasm64")))]
 #[inline]
 fn get_granularity_helper() -> usize {
     // <https://webassembly.github.io/spec/core/exec/runtime.html#page-size>

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -107,6 +107,17 @@ mod unix {
     }
 }
 
+// WebAssembly section
+
+// WebAssembly does not have a specific allocation granularity.
+// The page size works well.
+#[cfg(any(target_arch = "wasm32", target_arch = "wasm64"))]
+#[inline]
+fn get_granularity_helper() -> usize {
+    // <https://webassembly.github.io/spec/core/exec/runtime.html#page-size>
+    65536
+}
+
 // Windows Section
 
 #[cfg(all(windows, feature = "no_std"))]

